src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs

```diff
@@ -98,15 +108,15 @@ private static void PrependFoldersToEnvironmentPath()
                 var folders = new List<string>();
                 folders.Add(Path.Combine(home, @"site\tools"));
 
-                string path = Environment.GetEnvironmentVariable("PATH");
+                string path = _settingsManager.GetSetting("PATH");
                 string additionalPaths = String.Join(";", folders);
 
                 // Make sure we haven't already added them. This can happen if the appdomain restart (since it's still same process)
                 if (!path.Contains(additionalPaths))
                 {
                     path = additionalPaths + ";" + path;
 
-                    Environment.SetEnvironmentVariable("PATH", path);
+                    _settingsManager.SetSetting("PATH", path);
```

Comment:
This will not work with the current implementation of the settings manager as it does not update the actual environment variable (updates the dictionary only). This should either remain using the environment here or the manager should be modified to update the underlying store/source, by setting the environment variables (recommend option #  2 here)